qa/cephfs: add test for ACLs

rishabh-d-dave opened the pull request:
<!--
Thank you for opening a pull request!  Here are some tips on creating
a well formatted contribution.

Please give your pull request a title like "[component]: [short description]"

This is the format for commit messages:

"""
[component]: [short description]

[A longer multiline description]

Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
Signed-off-by: [Your Name] <[your email]>
"""

The Signed-off-by line is important, and it is your certification that
your contributions satisfy the Developers Certificate or Origin.  For
more detail, see SubmittingPatches.rst.

The component is the short name of a major daemon or subsystem,
something like "mon", "osd", "mds", "rbd, "rgw", etc. For ceph-mgr modules,
give the component as "mgr/<module name>" rather than a path into pybind.

For more examples, simply use "git log" and look at some historical commits.

This was just a quick overview.  More information for contributors is available here:
https://raw.githubusercontent.com/ceph/ceph/master/SubmittingPatches.rst

-->

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

jtlayton commented on qa/tasks/cephfs/test_acls.py:
```diff
@@ -0,0 +1,104 @@
+                export TEST_DEV={}
+                export TEST_DIR={}
+                #export SCRATCH_DEV={}
+                #export SCRATCH_MNT={}
```
This will test kcephfs. Are you also looking to run tests vs. ceph-fuse?

rishabh-d-dave replied:
Yep. But if this PR is really good I can start testing it on teuthology and get the test working for FUSE simultaneously on a separate PR so that this gets done faster.

What about `SCRATCH_DEV` and `SCRATCH_MNT`? They are commented right now because I can't get the tests running with them uncommented.

rishabh-d-dave commented:
> Looks like a reasonable starting point. I think you probably want to be able to run this vs. both kcephfs and ceph-fuse, so we can compare results between them.

Yeah, I intend to run both. But I am not clear on comparison part. Comparison in terms of functionality or performance?

rishabh-d-dave commented:
Major stuff about last push -
* got the tests working fine for FUSE mounts.
* added [`qa/suites/fs/basic_functional/tasks/acls.yaml`](https://github.com/rishabh-d-dave/ceph/blob/add-test-for-acls/qa/suites/fs/basic_functional/tasks/acls.yaml) so that I can start testing the FUSE part using teuthology.
* added methods that would test ACLs on kernel and FUSE mounts.

If I looked well around all the helper test code, there's no way to get a kernel mount on vstart cluster. So, I am trying to get it working.

@jtlayton @batrick 
Is it good to have separate methods to test ACLs on kernel and FUSE mounts? I mean, is there a better way to do this?

rishabh-d-dave commented:
The teuthology test failed because I need to add a method to teuthology - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-07_08:50:31-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

```
2019-08-07T09:22:03.056 INFO:tasks.cephfs_test_runner:======================================================================
2019-08-07T09:22:03.056 INFO:tasks.cephfs_test_runner:ERROR: test_acls_on_fuse_mount (tasks.cephfs.test_acls.TestACLs)
2019-08-07T09:22:03.056 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-08-07T09:22:03.056 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-08-07T09:22:03.056 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-add-test-for-acls/qa/tasks/cephfs/test_acls.py", line 12, in setUp
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:    self.prepare_xfstests_dev()
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-add-test-for-acls/qa/tasks/cephfs/test_acls.py", line 23, in prepare_xfstests_dev
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:    self.get_xfstests_dev()
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-add-test-for-acls/qa/tasks/cephfs/test_acls.py", line 45, in get_xfstests_dev
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:    self.repo_path = join(self.mount_a.client_remote.mkdtemp(), 'xfstests-dev')
2019-08-07T09:22:03.057 INFO:tasks.cephfs_test_runner:AttributeError: 'Remote' object has no attribute 'mkdtemp'

```

I've raised a PR for it - https://github.com/ceph/teuthology/pull/1301

jtlayton commented:
> @jtlayton @batrick
> Is it good to have separate methods to test ACLs on kernel and FUSE mounts? I mean, is there a better way to do this?

Unfortunately here's where we bump up against the limits of my knowledge of teuthology. I think what you want to aim for is to be able to run the same exact test on a kernel mount vs. ceph-fuse mount. i.e., have the only real difference be in how the mount is done.

rishabh-d-dave commented:
> > @jtlayton @batrick
> > Is it good to have separate methods to test ACLs on kernel and FUSE mounts? I mean, is there a better way to do this?
> 
> Unfortunately here's where we bump up against the limits of my knowledge of teuthology. I think what you want to aim for is to be able to run the same exact test on a kernel mount vs. ceph-fuse mount. i.e., have the only real difference be in how the mount is done.

Yep, I am trying just that. :)

rishabh-d-dave commented:
@batrick To get the tests running with the kernel mount, I've written this patch; but, it doesn't work the way I had expected it to - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-10_17:20:17-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

rishabh-d-dave commented:
@batrick 

I see following traceback on a teuthology test -

```
2019-08-13T13:59:22.355 INFO:tasks.cephfs_test_runner:======================================================================
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:ERROR: test_acls (tasks.cephfs.test_acls.TestACLs)
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-add-test-for-acls/qa/tasks/cephfs/xfstests_dev.py", line 7, in setUp
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:    self.prepare_xfstests_dev()
2019-08-13T13:59:22.356 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-rishabh-add-test-for-acls/qa/tasks/cephfs/xfstests_dev.py", line 16, in prepare_xfstests_dev
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:    self.mount_a.client_remote.run(args=['make'], cwd=self.repo_path)
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-rishabh-remote-add-mkdtemp/teuthology/orchestra/remote.py", line 205, in run
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-rishabh-remote-add-mkdtemp/teuthology/orchestra/run.py", line 437, in run
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:    r.wait()
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-rishabh-remote-add-mkdtemp/teuthology/orchestra/run.py", line 162, in wait
2019-08-13T13:59:22.357 INFO:tasks.cephfs_test_runner:    self._raise_for_status()
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_wip-rishabh-remote-add-mkdtemp/teuthology/orchestra/run.py", line 184, in _raise_for_status
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:    node=self.hostname, label=self.label
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:CommandFailedError: Command failed on smithi183 with status 2: '(cd /tmp/tmp.OisfoO5IWu/xfstests-dev && exec make)'
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-08-13T13:59:22.358 INFO:tasks.cephfs_test_runner:Ran 1 test in 70.954s
```

Any ideas what lead to addition of `exec` in the arguments of the command `cd /tmp/tmp.OisfoO5IWu/xfstests-dev && exec make`? Doesn't happen when I run this locally - 
```
2019-08-13 19:43:27,319.319 INFO:__main__:Running ['make']
2019-08-13 19:44:03,693.693 INFO:__main__:Running ['sudo', 'make', 'install']
```

batrick commented:
> @batrick Based on our conversation (I hope I understood it right), I wrote [this patch](https://github.com/ceph/ceph/pull/29421/commits/c24e30dd9767426bb05b8262b3e300ce05ccf052#diff-ecd3f347391c229cca2b4ecc64ac161bR1-R4) to get the tests running with the kernel mount but, it doesn't work the way I had expected it to - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-10_17:20:17-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

You shouldn't put the test in kcephfs/recovery. Try kcephfs/cephfs. The reason for those failures mostly appears to be because there are too many clients (4) so teuthology can't schedule the job.

batrick commented:
@rishabh-d-dave the `exec` is probably added by teuthology but I don't immediately see where. I don't think that by itself is the reason for your error. You'll need to check the stderr of the command. Please also share the teuthology log that snippet is from.

rishabh-d-dave commented:
> > @batrick Based on our conversation (I hope I understood it right), I wrote [this patch](https://github.com/ceph/ceph/pull/29421/commits/c24e30dd9767426bb05b8262b3e300ce05ccf052#diff-ecd3f347391c229cca2b4ecc64ac161bR1-R4) to get the tests running with the kernel mount but, it doesn't work the way I had expected it to - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-10_17:20:17-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/
> 
> You shouldn't put the test in kcephfs/recovery. Try kcephfs/cephfs. The reason for those failures mostly appears to be because there are too many clients (4) so teuthology can't schedule the job.

I did put the test in kcephfs/cephfs earlier but I got a similar result to that of kcephfs/recovery.

rishabh-d-dave commented:
> @rishabh-d-dave the `exec` is probably added by teuthology but I don't immediately see where. I don't think that by itself is the reason for your error. You'll need to check the stderr of the command. Please also share the teuthology log that snippet is from.

http://pulpito.front.sepia.ceph.com/rishabh-2019-08-13_06:38:57-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

batrick commented:
http://pulpito.front.sepia.ceph.com/rishabh-2019-08-13_06:38:57-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/4212321/

is failing because it's trying to use `apt` on a rhel machine.

batrick commented:
`exec make` is not the issue here. I think you're missing a dependency but I'm not sure. I recommend using `teuthology-lock` to get a rhel7 machine and then try runnign the build yourself to see what's missing/wrong.

rishabh-d-dave commented:
@batrick 
dump, indent, libaio-devel and btrfs-prog-devel couldn't be installed on the teuthology machine I could obtain from locking; these packages aren't available.

batrick commented:
> > > @batrick Based on our conversation (I hope I understood it right), I wrote [this patch](https://github.com/ceph/ceph/pull/29421/commits/c24e30dd9767426bb05b8262b3e300ce05ccf052#diff-ecd3f347391c229cca2b4ecc64ac161bR1-R4) to get the tests running with the kernel mount but, it doesn't work the way I had expected it to - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-10_17:20:17-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/
> > 
> > 
> > You shouldn't put the test in kcephfs/recovery. Try kcephfs/cephfs. The reason for those failures mostly appears to be because there are too many clients (4) so teuthology can't schedule the job.
> 
> I did put the test in kcephfs/cephfs earlier but I got a similar result to that of kcephfs/recovery.

The intent of those sub-suites is different. Recovery tests are going to exercise failures in various ways. The kcephfs:cephfs suite is more generic testing and would be appropriate for ACL tests.

rishabh-d-dave commented:
> > > > @batrick Based on our conversation (I hope I understood it right), I wrote [this patch](https://github.com/ceph/ceph/pull/29421/commits/c24e30dd9767426bb05b8262b3e300ce05ccf052#diff-ecd3f347391c229cca2b4ecc64ac161bR1-R4) to get the tests running with the kernel mount but, it doesn't work the way I had expected it to - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-10_17:20:17-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/
> > > 
> > > 
> > > You shouldn't put the test in kcephfs/recovery. Try kcephfs/cephfs. The reason for those failures mostly appears to be because there are too many clients (4) so teuthology can't schedule the job.
> > 
> > 
> > I did put the test in kcephfs/cephfs earlier but I got a similar result to that of kcephfs/recovery.
> 
> The intent of those sub-suites is different. Recovery tests are going to exercise failures in various ways. The kcephfs:cephfs suite is more generic testing and would be appropriate for ACL tests.

FUSE only triggers 4 jobs but kernel triggers 22 jobs - http://pulpito.front.sepia.ceph.com/rishabh-2019-08-28_15:40:40-kcephfs-wip-rishabh-add-test-for-acls-distro-basic-smithi/. Is that normal?

I used similar command to trigger both -

```
$ teuthology-suite -v -m smithi -c wip-rishabh-add-test-for-acls -s fs -p 50 --filter "acls-fuse-clients"
$ teuthology-suite -v -m smithi -c wip-rishabh-add-test-for-acls -s kcephfs -p 50 --filter acls-kernel-client
```

rishabh-d-dave commented:
...changed the SHA of the last commit since centos builds on shaman failed.

rishabh-d-dave commented:
about last push, did a rebase.

rishabh-d-dave commented:
ACL tests passed with FUSE mounts on teuthology - http://pulpito.front.sepia.ceph.com/rishabh-2019-09-18_05:15:15-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

```
2019-09-18T05:37:01.499 INFO:teuthology.orchestra.run.smithi148:> (cd /tmp/tmp.Ju9rLCNDPe/xfstests-dev && exec sudo ./check generic/099)
2019-09-18T05:37:01.832 INFO:teuthology.orchestra.run.smithi148.stdout:FSTYP         -- ceph
2019-09-18T05:37:01.832 INFO:teuthology.orchestra.run.smithi148.stdout:PLATFORM      -- Linux/x86_64 smithi148 3.10.0-1062.el7.x86_64 #1 SMP Thu Jul 18 20:25:13 UTC 2019
2019-09-18T05:37:01.832 INFO:teuthology.orchestra.run.smithi148.stdout:
2019-09-18T05:37:02.511 INFO:teuthology.orchestra.run.smithi148.stdout:generic/099       1s
2019-09-18T05:37:02.511 INFO:teuthology.orchestra.run.smithi148.stdout:Ran: generic/099
2019-09-18T05:37:02.512 INFO:teuthology.orchestra.run.smithi148.stdout:Passed all 1 tests
2019-09-18T05:37:02.512 INFO:teuthology.orchestra.run.smithi148.stdout:
2019-09-18T05:37:06.256 INFO:teuthology.orchestra.run.smithi148:Running:
```

rishabh-d-dave commented:
Rebased the PR.

jtlayton commented on qa/tasks/cephfs/kernel_mount.py:
```diff
@@ -66,7 +66,7 @@ def mount(self, mount_path=None, mount_fs_name=None, mountpoint=None):
             ],
-            timeout=(5*60),
+            timeout=(5*60), check_status=False
         )
```
I'm not sure about this -- the mount command could fail for reasons besides it being already mounted. Perhaps you should just check whether it's mounted before issuing the mount command? Or maybe consider unmounting and not checking the error and then redoing the mount?

rishabh-d-dave replied:
Yep, something like that would be better.

rishabh-d-dave replied:
On a proper look at the above snippet -

```
        self.client_remote.run(
            args=[
                'mkdir',
                '--',
                self.mountpoint,
            ],
            timeout=(5*60), check_status=False
)
```

Isn't this actually change fine? It just creates the mountpoint directory and doesn't mount anything. I recall, now, that this change was made because if the mountpoint directory is already created, the method would crash and the test would abort. I've disabled just this crash/abort part.

jtlayton replied:
Yes, I misread. I was thinking this was the mount command. For the mkdir it's  probably fine. That said, the usual way to deal with "create the directory if it doesn't already exist" is to use "mkdir -p".

rishabh-d-dave replied:
"mkdir -p" would be better.

rishabh-d-dave replied:
Made the change - https://github.com/ceph/ceph/pull/29421/files#diff-42931b13153c8474d563f8a82f789b51R55.

Thanks!

rishabh-d-dave commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,171 @@
+        # superuser since the repo is cloned somewhere in /tmp.
+        self.mount_a.client_remote.run(args=['sudo', 'make'],
+                                       cwd=self.repo_path, stdout=StringIO(),
+                                       stderr=StringIO())
```
On teuthology machines it requires superuser access to operate on directories in /tmp.

rishabh-d-dave replied:
> make install almost certainly does though. That said, I generally never call make install with xfstests and just run it out of the build tree, so YMMV.

The make install part isn't clear to me. Can you please elaborate?

jtlayton replied:
You're doing a "make" and then a "make install". The "make" typically doesn't require any special privileges since you're just operating in the build dir, but the "make install" can drop files all over the filesystem, so you often need root privileges for that.

You can also run xfstests directly out of the tree without running "make install", but now that I look, I don't see directions for that in the README.

rishabh-d-dave replied:
> You can also run xfstests directly out of the tree without running "make install", but now that I look, I don't see directions for that in the README.

Would be nice we could run without doing "make install".

rishabh-d-dave commented on qa/tasks/vstart_runner.py:
```diff
@@ -307,7 +307,7 @@ def _perform_checks_and_return_list_of_args(self, args, omit_sudo):
     def run(self, args, check_status=True, wait=True,
             stdout=None, stderr=None, cwd=None, stdin=None,
-            logger=None, label=None, env=None, timeout=None, omit_sudo=True):
+            logger=None, label=None, env=None, timeout=None, omit_sudo=False):
```
Because nobody would use `sudo`and intend to get rid of it from the command arguments before it's execution. Also, teuthology's run already has `omit_sudo` set to `False`. So this should be safe. Teuthology's run - https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/run.py#L381

jtlayton replied:
Fair enough. Maybe we should just get rid of that parameter altogether? It doesn't seem to make a lot of sense, but maybe I'm missing something.

rishabh-d-dave replied:
Yes. Now that I am well-acquainted with vstart_runner.py, I think the same.

When I added that parameter, there was a comment that said "we don't need stinkin' sudo" due to which I wasn't sure why not. I'll do this on a different PR.

rishabh-d-dave commented:
Testing with teuthology was successful after rebase too. However, I had to kill jobs running for FUSE mounts myself and ones for kernel mounts crashed after tests and it's teardown ran successfully. The problem is that there is an attempt to delete mountpoint directory after test's teardown has deleted it already. This leads to a crash.

Added a fix for that in last push - https://github.com/ceph/ceph/pull/29421/commits/073a64cd16d8b6eb3df30c88e5ac7f46acfda4de

jtlayton commented on qa/tasks/cephfs/test_acls.py:
```diff
@@ -0,0 +1,22 @@
+        self.mount_a.client_remote.run(args=['sudo', './check',
+            'generic/099'], cwd=self.repo_path, stdout=StringIO(),
+            stderr=StringIO(), timeout=30, check_status=True,
+            label='running tests for ACLs from xfstests-dev')
```
Huh...now that we have all of this infrastructure here, I think we ought to expand this to allow for a full xfstests run. It seems a shame to do all of this setup just to run a single test.

That said, a few tests are known to fail on cephfs (at least for now), so I think we'd need to either whitelist the failures or skip them.

That can be done in a later PR though. I'd suggest that we just merge this as an ACL test only for now and look at doing that later.

rishabh-d-dave replied:
> That can be done in a later PR though. I'd suggest that we just merge this as an ACL test only for now and look at doing that later.

Agreed. I'll create a PR that'll do that as soon as this one merges.

rishabh-d-dave commented:
Minor change requested here - https://github.com/ceph/ceph/pull/29421#discussion_r327049613.

rishabh-d-dave commented:
Since the tests are running fine against teuthology I'll defer testing until there's a major change on this PR or until the PR is approved.

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,171 @@
+        self.repo_path = join(self.mount_a.client_remote.mkdtemp(),
+                              'xfstests-dev')
+        self.mount_a.run_shell('%s %s %s' % ('git clone', repourl,
+                                             self.repo_path))
```
We can't be hitting `git.kernel.org` for every test we run. The right thing to do here is to export a tarball of some tagged version of xfstests-dev and hand that off to @djgalloway to put somewhere on ceph.com. Then just pull that down in the tests and build.

djgalloway replied:
I'm guessing that's why we have xfs-tests.git on git.ceph.com: http://git.ceph.com/?p=xfstests.git;a=summary

Shall I just add xfstests-dev?

batrick replied:
Works for me!

@rishabh-d-dave see `qa/tasks/restart.py` for how we download a tarball from that remote. There's no reason to clone the remote.

djgalloway replied:
http://git.ceph.com/?p=xfstests-dev.git;a=summary

batrick replied:
@rishabh-d-dave

rishabh-d-dave replied:
Cloning from `git.ceph.com`, now. Putting this to test on teuthology next.

Would've been nicer if I knew this beforehand since I had problems with cloning from `git.kernel.org`.

rishabh-d-dave replied:
@djgalloway Thanks!

rishabh-d-dave commented:
Rebased.

rishabh-d-dave commented:
As usual, FUSE tests passed but not without hanging - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-04_12:43:36-fs-add-test-for-acls-distro-basic-smithi/

batrick commented on qa/suites/fs/basic_functional/tasks/acls-fuse-client.yaml:
```diff
@@ -0,0 +1,8 @@
+tasks:
+  - install:
```
Oops, this doesn't work; my fault for recommending it. The install task does a full "unwind" even if we are only really trying to install system packages. This means that once the test finishes the install task will remove all of the packages.

Let's just move the packages to `qa/cephfs/begin.yaml`. I don't think there is an easier way to do it without changing a lot of code.

rishabh-d-dave replied:
Shouldn't the natural order for teardown be to stop anything "Ceph" first and then remove related files and then uninstall packages?

rishabh-d-dave replied:
> Let's just move the packages to qa/cephfs/begin.yaml. I don't think there is an easier way to do it without changing a lot of code.

Made the change.

rishabh-d-dave replied:
Was this in context to my last comment? I don't get the context, otherwise.

batrick replied:
> Shouldn't the natural order for teardown be to stop anything "Ceph" first and then remove related files and then uninstall packages?

Yes but you now have two `install` tasks:

````
  - install:
      extra_packages:
        deb:
        - python3-cephfs
        - cephfs-shell
        rpm:
        - python3-cephfs
      extra_system_packages:
        deb:
        - bison
        - flex
        - libelf-dev
        - libssl-dev
        rpm:
        - bison
        - flex
        - elfutils-libelf-devel
        - openssl-devel
  - ceph: null
  - ceph-fuse: null
  - install:
      extra_system_packages:
        deb:
        - libaio-dev
        rpm:
        - dump
        - indent
        - libaio-devel
        - btrfs-progs-devel
````

From: /ceph/teuthology-archive/rishabh-2019-10-04_12:43:36-fs-add-test-for-acls-distro-basic-smithi/4357983/teuthology.log

the second install task runs after the `ceph` task.

rishabh-d-dave commented:
Tested with FUSE, test passed but hanged as usual - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-09_10:18:26-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

batrick commented:
The new problem can be seen by this symptom:

````
2019-10-09T10:41:07.407 INFO:teuthology.task.internal:Tidying up after the test...
2019-10-09T10:41:07.408 INFO:teuthology.orchestra.run.smithi001:Running:
2019-10-09T10:41:07.408 INFO:teuthology.orchestra.run.smithi001:> find /home/ubuntu/cephtest -ls ; rmdir -- /home/ubuntu/cephtest
2019-10-09T10:41:07.412 INFO:teuthology.orchestra.run.smithi016:Running:
2019-10-09T10:41:07.412 INFO:teuthology.orchestra.run.smithi016:> find /home/ubuntu/cephtest -ls ; rmdir -- /home/ubuntu/cephtest
2019-10-09T10:41:07.451 INFO:teuthology.orchestra.run.smithi016.stdout:393219    4 drwxr-xr-x   2 ubuntu   ubuntu       4096 Oct  9 10:41 /home/ubuntu/cephtest
2019-10-09T12:45:08.307 DEBUG:teuthology.exit:Got signal 15; running 2 handlers...
2019-10-09T12:45:08.314 DEBUG:teuthology.task.console_log:Killing console logger for smithi016
2019-10-09T12:45:08.315 DEBUG:teuthology.task.console_log:Killing console logger for smithi001
2019-10-09T12:45:08.315 DEBUG:teuthology.task.console_log:Killing console logger for smithi016
2019-10-09T12:45:08.315 DEBUG:teuthology.task.console_log:Killing console logger for smithi001
````

One of the `find` commands is hanging. This generally indicates an unresponsive mount point. In this case, you should suspect ceph-fuse is hanging around.

I would look here:

````
2019-10-09T10:39:14.763 INFO:teuthology.orchestra.run.smithi001:Running:
2019-10-09T10:39:14.763 INFO:teuthology.orchestra.run.smithi001:> sudo umount /tmp/tmp.4XY7w0sfMxtest /tmp/tmp.YCusnX5gbxscratch
2019-10-09T10:39:14.830 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi001.stderr:ceph-fuse[24260]: fuse finished with error 0 and tester_r 0
2019-10-09T10:39:14.853 INFO:tasks.cephfs.fuse_mount.ceph-fuse.0.smithi001.stderr:ceph-fuse[24817]: fuse finished with error 0 and tester_r 0
2019-10-09T10:39:14.854 INFO:teuthology.orchestra.run.smithi001:Running:
2019-10-09T10:39:14.854 INFO:teuthology.orchestra.run.smithi001:> sudo rm -rf /tmp/tmp.ejdxlGZOmMxfstests-dev /tmp/tmp.YCusnX5gbxscratch /tmp/tmp.4XY7w0sfMxtest
2019-10-09T10:39:14.918 INFO:tasks.cephfs_test_runner:test_acls (tasks.cephfs.test_acls.TestACLs) ... ok
2019-10-09T10:39:14.918 INFO:tasks.cephfs_test_runner:
2019-10-09T10:39:14.918 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-10-09T10:39:14.918 INFO:tasks.cephfs_test_runner:Ran 1 test in 109.574s
2019-10-09T10:39:14.918 INFO:tasks.cephfs_test_runner:
2019-10-09T10:39:14.919 INFO:tasks.cephfs_test_runner:OK
2019-10-09T10:39:14.919 DEBUG:teuthology.run_tasks:Unwinding manager cephfs_test_runner
2019-10-09T10:39:14.931 DEBUG:teuthology.run_tasks:Unwinding manager ceph-fuse
2019-10-09T10:39:14.944 INFO:tasks.ceph_fuse:Unmounting ceph-fuse clients...
2019-10-09T10:39:14.944 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-10-09T10:39:14.944 INFO:teuthology.orchestra.run.smithi001:Running:
2019-10-09T10:39:14.945 INFO:teuthology.orchestra.run.smithi001:> stat --file-system '--printf=%T
2019-10-09T10:39:14.945 INFO:teuthology.orchestra.run.smithi001:> ' -- /home/ubuntu/cephtest/mnt.1
2019-10-09T10:39:15.001 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-10-09T10:39:15.029 INFO:teuthology.orchestra.run.smithi001.stderr:stat: cannot read file system information for ‘/home/ubuntu/cephtest/mnt.1’: No such file or directory
2019-10-09T10:39:15.029 INFO:tasks.cephfs.fuse_mount:mount point does not exist: /home/ubuntu/cephtest/mnt.1
2019-10-09T10:39:15.029 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-10-09T10:39:15.030 INFO:teuthology.orchestra.run.smithi001:Running:
2019-10-09T10:39:15.030 INFO:teuthology.orchestra.run.smithi001:> stat --file-system '--printf=%T
2019-10-09T10:39:15.030 INFO:teuthology.orchestra.run.smithi001:> ' -- /tmp/tmp.YCusnX5gbxscratch
2019-10-09T10:39:15.114 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-10-09T10:39:15.143 INFO:teuthology.orchestra.run.smithi001.stderr:stat: cannot read file system information for ‘/tmp/tmp.YCusnX5gbxscratch’: No such file or directory
2019-10-09T10:39:15.143 INFO:tasks.cephfs.fuse_mount:mount point does not exist: /tmp/tmp.YCusnX5gbxscratch
2019-10-09T10:39:15.143 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-10-09T10:39:15.144 INFO:teuthology.orchestra.run.smithi016:Running:
2019-10-09T10:39:15.144 INFO:teuthology.orchestra.run.smithi016:> stat --file-system '--printf=%T
2019-10-09T10:39:15.144 INFO:teuthology.orchestra.run.smithi016:> ' -- /home/ubuntu/cephtest/mnt.3
2019-10-09T10:39:15.185 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-10-09T10:39:15.188 INFO:teuthology.orchestra.run.smithi016.stderr:stat: cannot read file system information for ‘/home/ubuntu/cephtest/mnt.3’: No such file or directory
2019-10-09T10:39:15.188 INFO:tasks.cephfs.fuse_mount:mount point does not exist: /home/ubuntu/cephtest/mnt.3
2019-10-09T10:39:15.189 INFO:teuthology.orchestra.run:Running command with timeout 900
2019-10-09T10:39:15.189 INFO:teuthology.orchestra.run.smithi016:Running:
2019-10-09T10:39:15.189 INFO:teuthology.orchestra.run.smithi016:> stat --file-system '--printf=%T
2019-10-09T10:39:15.189 INFO:teuthology.orchestra.run.smithi016:> ' -- /home/ubuntu/cephtest/mnt.2
2019-10-09T10:39:15.293 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-10-09T10:39:15.323 INFO:teuthology.orchestra.run.smithi016.stderr:stat: cannot read file system information for ‘/home/ubuntu/cephtest/mnt.2’: No such file or directory
2019-10-09T10:39:15.324 INFO:tasks.cephfs.fuse_mount:mount point does not exist: /home/ubuntu/cephtest/mnt.2
````

Looks like ceph-fuse task is trying to unmount /tmp/tmp.YCusnX5gbxscratch but it was already unmounted and removed. Consider that you are overriding the existing mount point for `mount_a` when you mount that tmp directory. The old ceph-fuse instance is probably left behind and not cleaned up!

batrick commented:
@rishabh-d-dave please post test results when done.

rishabh-d-dave commented:
Yes, yes. Definitely.

rishabh-d-dave commented:
Passed but hanged - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-11_05:46:42-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

I'll try to dig in and find out the cause.

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,174 @@
+        self.mount_a.run_shell(['mkdir', self.test_dirname,
+                                self.scratch_dirname])
+
+        self.mount_test = self.mount_b
```
I think you need to unmount b/c before you mount somewhere else.

rishabh-d-dave replied:
Done.

rishabh-d-dave commented:
Tests passed successfully - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-12_10:58:58-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,179 @@
+            suffix=self.scratch_dirname)
+
+        self.mount_a.run_shell(['mkdir', self.test_dirname,
+                                self.scratch_dirname])
```
Is it strictly necessary that `test_dirname` and `scratch_dirname` be in CephFS? Requiring three mount points (three clients) should be avoided. Alternatively, we can just use symlinks instead of two extra mounts?

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,179 @@
+            #export SCRATCH_MNT={}
+            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={}"
+            ''').format(self.test_dev, self.test_dirs_mount_path, self.scratch_dev,
+                        self.scratch_dirs_mount_path, self.get_admin_key())
```
Presumably `TEST_FS_MOUNT_OPS` is so xfs-tests can remount?

The monitor addresses should no longer be necessary now that mount.ceph gets the information from ceph.conf. See `kernel_mount.py` and recent changes made by Jeff.

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,179 @@
+
+
+class XFSTestsDev(CephFSTestCase):
+    CLIENTS_REQUIRED = 3
```
This is why the kclient tests fail. You are asking for 3 clients but only one is available:

````
pdonnell@aglarond ~/ceph/qa/suites/kcephfs/cephfs/clusters$ cat 1-mds-1-client.yaml 
roles:
- [mon.a, mgr.y, mds.a, mds.c, osd.0, osd.1, osd.2, osd.3]
- [mon.b, mon.c, mgr.x, mds.b, osd.4, osd.5, osd.6, osd.7]
- [client.0]
````

batrick replied:
If three clients were absolutely necessary, you would need to add two clients to the test by appending two clients to the `roles` array. (I think that's valid...)

rishabh-d-dave replied:
Yeah, I anticipated an issue like this but wasn't sure when I mentioned about this on standup.

batrick commented:
@rishabh-d-dave please update with new qa results when you're ready.

rishabh-d-dave commented:
Teuth tests using FUSE failed on Ubuntu due to packages but ones on CentOS passed -

http://pulpito.front.sepia.ceph.com/rishabh-2019-10-19_04:12:54-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

rishabh-d-dave commented:
FUSE-based teuth tests passed - http://pulpito.front.sepia.ceph.com/rishabh-2019-10-19_11:15:54-fs-wip-rishabh-add-test-for-acls-ubuntu-fix-distro-basic-smithi/

rishabh-d-dave commented:
Newly pushed commits are same as the one which is getting tested.

batrick commented:
Will run this through the next QA run to make sure the non-ACL changes don't break anything else.

batrick commented on qa/tasks/cephfs/xfstests_dev.py:
```diff
@@ -0,0 +1,172 @@
+                'mon_status'))
+            mon_socket_addr = mon_status['monmap']['mons'][0]['public_addrs']\
+                ['addrvec'][1]['addr']
+            return mon_socket_addr
```
Since c09b82a80a392ccd0da7677c7b424ce5cd3fa5d6 this command is now a tell/asok command.

Better is to use `ceph --format=json mon dump` to get the addresses. I recommend making this a helper method in `qa/tasks/ceph_manager.py`. cc @liewegas 

See also this failure: `/ceph/teuthology-archive/pdonnell-2019-11-04_12:55:15-fs-wip-pdonnell-testing-20191104.085350-distro-basic-smithi/4472363/teuthology.log`

After fixing, please rerun through teuthology and link the results in this PR.

rishabh-d-dave replied:
http://qa-proxy.ceph.com/teuthology/pdonnell-2019-11-04_12:55:15-fs-wip-pdonnell-testing-20191104.085350-distro-basic-smithi/4472363/teuthology.log

rishabh-d-dave commented:
Teuth tests for FUSE-based mounts passed - http://pulpito.front.sepia.ceph.com/rishabh-2019-11-06_14:41:05-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/

rishabh-d-dave commented:
Tests on teuth for FUSE mounts worked fine last time - http://pulpito.front.sepia.ceph.com/rishabh-2019-11-08_12:31:20-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/; perhaps the issue in last to last testing was only transient.

batrick commented:
Looks like you maybe found a bug?

````
2019-11-06T18:00:31.370223+00:00 smithi176 kernel: [  540.182303] libceph: client5187 fsid e1f67b45-1494-4a64-8fb0-a6d86270a4a3
2019-11-06T18:00:32.394242+00:00 smithi176 kernel: [  541.197923] cache_from_obj: Wrong slab cache. inode_cache but object is from ceph_inode_info
2019-11-06T18:00:32.394257+00:00 smithi176 kernel: [  541.206406] WARNING: CPU: 0 PID: 0 at /build/linux-WKYm23/linux-4.15.0/mm/slab.h:378 kmem_cache_free+0x142/0x1e0
2019-11-06T18:00:32.394259+00:00 smithi176 kernel: [  541.206407] Modules linked in: ceph libceph fscache xfs libcrc32c aufs overlay intel_rapl sb_edac x86_pkg_temp_thermal intel_powerclamp coretemp crct10dif_pclmul crc32_pclmul ghash_clmulni_intel pcbc aesni_intel aes_x86_64 crypto_simd glue_helper cryptd joydev input_leds ast intel_cstate ttm drm_kms_helper intel_rapl_perf drm fb_sys_fops syscopyarea sysfillrect wmi sysimgblt ipmi_ssif lpc_ich acpi_power_meter ipmi_si ipmi_devintf acpi_pad ipmi_msghandler ioatdma mei_me shpchp mac_hid mei kvm_intel kvm irqbypass sch_fq_codel ib_iser rdma_cm iw_cm ib_cm ib_core iscsi_tcp libiscsi_tcp libiscsi scsi_transport_iscsi lp parport nfsd auth_rpcgss nfs_acl lockd grace sunrpc ip_tables x_tables autofs4 btrfs xor zstd_compress raid6_pq hid_generic usbhid hid igb ixgbe i2c_algo_bit dca ptp ahci nvme libahci pps_core
2019-11-06T18:00:32.394263+00:00 smithi176 kernel: [  541.206454]  nvme_core mdio
2019-11-06T18:00:32.394274+00:00 smithi176 kernel: [  541.206458] CPU: 0 PID: 0 Comm: swapper/0 Not tainted 4.15.0-66-generic #75-Ubuntu
2019-11-06T18:00:32.394275+00:00 smithi176 kernel: [  541.206459] Hardware name: Supermicro SYS-5018R-WR/X10SRW-F, BIOS 2.0 12/17/2015
2019-11-06T18:00:32.394276+00:00 smithi176 kernel: [  541.206461] RIP: 0010:kmem_cache_free+0x142/0x1e0
2019-11-06T18:00:32.394277+00:00 smithi176 kernel: [  541.206462] RSP: 0018:ffff9b0ebfc03eb8 EFLAGS: 00010286
2019-11-06T18:00:32.394277+00:00 smithi176 kernel: [  541.206463] RAX: 0000000000000050 RBX: ffff9b0e6669ceb8 RCX: 0000000000000000
2019-11-06T18:00:32.394278+00:00 smithi176 kernel: [  541.206464] RDX: 0000000000000000 RSI: ffff9b0ebfc16498 RDI: ffff9b0ebfc16498
2019-11-06T18:00:32.394280+00:00 smithi176 kernel: [  541.206465] RBP: ffff9b0ebfc03ed0 R08: 0000000000000406 R09: 0000000000cdcdcd
2019-11-06T18:00:32.394280+00:00 smithi176 kernel: [  541.206466] R10: 000000000007a120 R11: 0000000000000007 R12: ffff9b0e9f27e780
2019-11-06T18:00:32.394281+00:00 smithi176 kernel: [  541.206467] R13: ffff9b0ebfc23678 R14: ffff9b0ebfc03f00 R15: 000000000000000a
2019-11-06T18:00:32.394281+00:00 smithi176 kernel: [  541.206468] FS:  0000000000000000(0000) GS:ffff9b0ebfc00000(0000) knlGS:0000000000000000
2019-11-06T18:00:32.394282+00:00 smithi176 kernel: [  541.206469] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
2019-11-06T18:00:32.394283+00:00 smithi176 kernel: [  541.206470] CR2: 0000564d12505000 CR3: 000000010240a003 CR4: 00000000003606f0
2019-11-06T18:00:32.394284+00:00 smithi176 kernel: [  541.206471] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
2019-11-06T18:00:32.394285+00:00 smithi176 kernel: [  541.206472] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
2019-11-06T18:00:32.394286+00:00 smithi176 kernel: [  541.206473] Call Trace:
2019-11-06T18:00:32.394286+00:00 smithi176 kernel: [  541.206474]  <IRQ>
2019-11-06T18:00:32.394287+00:00 smithi176 kernel: [  541.206480]  i_callback+0x1c/0x20
2019-11-06T18:00:32.394287+00:00 smithi176 kernel: [  541.206484]  rcu_process_callbacks+0x1a7/0x4c0
2019-11-06T18:00:32.394288+00:00 smithi176 kernel: [  541.206486]  __do_softirq+0xe4/0x2d4
2019-11-06T18:00:32.394290+00:00 smithi176 kernel: [  541.206491]  irq_exit+0xc5/0xd0
2019-11-06T18:00:32.394290+00:00 smithi176 kernel: [  541.206493]  smp_apic_timer_interrupt+0x79/0x140
2019-11-06T18:00:32.394291+00:00 smithi176 kernel: [  541.206495]  apic_timer_interrupt+0x8c/0xa0
2019-11-06T18:00:32.394291+00:00 smithi176 kernel: [  541.206496]  </IRQ>
2019-11-06T18:00:32.394292+00:00 smithi176 kernel: [  541.206500] RIP: 0010:cpuidle_enter_state+0xa7/0x2f0
2019-11-06T18:00:32.394292+00:00 smithi176 kernel: [  541.206501] RSP: 0018:ffffffff8d803e10 EFLAGS: 00000246 ORIG_RAX: ffffffffffffff11
2019-11-06T18:00:32.394292+00:00 smithi176 kernel: [  541.206502] RAX: ffff9b0ebfc22840 RBX: 0000007e01e8c78e RCX: 000000000000001f
2019-11-06T18:00:32.394294+00:00 smithi176 kernel: [  541.206503] RDX: 0000007e01e8c78e RSI: ffffffc5b7e971f8 RDI: 0000000000000000
2019-11-06T18:00:32.394295+00:00 smithi176 kernel: [  541.206504] RBP: ffffffff8d803e50 R08: 0000000000000004 R09: 0000000000022080
2019-11-06T18:00:32.394295+00:00 smithi176 kernel: [  541.206505] R10: ffffffff8d803de0 R11: 000002851da38465 R12: ffff9b0ebfc2d950
2019-11-06T18:00:32.394296+00:00 smithi176 kernel: [  541.206505] R13: 0000000000000004 R14: ffffffff8d9733d8 R15: 0000000000000000
2019-11-06T18:00:32.394296+00:00 smithi176 kernel: [  541.206508]  cpuidle_enter+0x17/0x20
2019-11-06T18:00:32.394297+00:00 smithi176 kernel: [  541.206512]  call_cpuidle+0x23/0x40
2019-11-06T18:00:32.394298+00:00 smithi176 kernel: [  541.206514]  do_idle+0x18c/0x1f0
2019-11-06T18:00:32.394299+00:00 smithi176 kernel: [  541.206515]  cpu_startup_entry+0x73/0x80
2019-11-06T18:00:32.394299+00:00 smithi176 kernel: [  541.206518]  rest_init+0xae/0xb0
2019-11-06T18:00:32.394300+00:00 smithi176 kernel: [  541.206523]  start_kernel+0x4dc/0x4fd
2019-11-06T18:00:32.394300+00:00 smithi176 kernel: [  541.206525]  x86_64_start_reservations+0x24/0x26
2019-11-06T18:00:32.394301+00:00 smithi176 kernel: [  541.206526]  x86_64_start_kernel+0x74/0x77
2019-11-06T18:00:32.394301+00:00 smithi176 kernel: [  541.206529]  secondary_startup_64+0xa5/0xb0
````

From: /ceph/teuthology-archive/rishabh-2019-11-06_17:44:41-fs-wip-rishabh-add-test-for-acls-distro-basic-smithi/4477008/remote/smithi176/syslog/kern.log.gz

ping @jtlayton

rishabh-d-dave commented:
> Looks like you maybe found a bug?

I guess, yes. But earlier tests on FUSE mounts have run with teuthology successfully.

jtlayton commented:
> Looks like you maybe found a bug?
>
> ping @jtlayton

Yes, this is almost certainly due to the ubuntu kernels sucking in the bad backport from the stable series and is unrelated to anything in this PR.

I wonder if there is a way we can "blacklist" certain ubuntu kernels for kclient testing? Or just not allow ubuntu clients in that role until the fixes are in?

batrick commented:
> > Looks like you maybe found a bug?
> > ping @jtlayton
> 
> Yes, this is almost certainly due to the ubuntu kernels sucking in the bad backport from the stable series and is unrelated to anything in this PR.
> 
> I wonder if there is a way we can "blacklist" certain ubuntu kernels for kclient testing? Or just not allow ubuntu clients in that role until the fixes are in?

Okay so this is a puzzle. The configuration for the test should be a ceph-fuse client and yet a kernel client was mounted. That's evidenced by the `kern.log.gz` above, around 2019-11-06T18:00:31. In the teuthology log around that time:

> 2019-11-06T18:00:31.008 INFO:teuthology.orchestra.run.smithi176:Running (running tests for ACLs from xfstests-dev):
> 2019-11-06T18:00:31.009 INFO:teuthology.orchestra.run.smithi176:> (cd /tmp/tmp.boVX29DprWxfstests-dev && exec sudo ./check generic/099)

I don't see why that would mount the kernel. @rishabh-d-dave do you know what's going on?

rishabh-d-dave commented:
Whether the mount is FUSE or kernel based depends on teuthology's suite. And the issue came up in fs suite which is always(?) gets a FUSE mount. So, I don't see how can my code get a kernel mount here.

I'll go through the log and code again to see if I missed some detail.

rishabh-d-dave commented:
FS was surely remounted by xfstests-dev (if crashed was surely due to kernel mount). The line `tasks.cephfs.test_acls:client is fuse mounted` is printed just before running the test in the following snippet -

```
2019-11-06T16:24:40.937 INFO:teuthology.orchestra.run.smithi148.stdout:../install-sh -o root -g root -m 755 -d /var/lib/xfstests/tests
2019-11-06T16:24:40.956 INFO:teuthology.orchestra.run.smithi148.stdout:./install-sh -o root -g root -m 755 -d /var/lib/xfstests
2019-11-06T16:24:40.974 INFO:teuthology.orchestra.run.smithi148.stdout:./install-sh -o root -g root -m 755 check /var/lib/xfstests
2019-11-06T16:24:40.995 INFO:teuthology.orchestra.run.smithi148.stdout:./install-sh -o root -g root -m 644 randomize.awk /var/lib/xfstests
2019-11-06T16:24:41.017 INFO:tasks.cephfs.test_acls:client is fuse mounted
2019-11-06T16:24:41.017 INFO:teuthology.orchestra.run:Running command with timeout 30
2019-11-06T16:24:41.017 INFO:teuthology.orchestra.run.smithi148:Running (running tests for ACLs from xfstests-dev):
2019-11-06T16:24:41.017 INFO:teuthology.orchestra.run.smithi148:> (cd /tmp/tmp.GPZijsyn3Pxfstests-dev && exec sudo ./check generic/099)
2019-11-06T16:24:41.522 INFO:teuthology.orchestra.run.smithi148.stdout:FSTYP         -- ceph
2019-11-06T16:24:41.522 INFO:teuthology.orchestra.run.smithi148.stdout:PLATFORM      -- Linux/x86_64 smithi148 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019
2019-11-06T16:24:41.522 INFO:teuthology.orchestra.run.smithi148.stdout:
2019-11-06T16:24:42.806 INFO:teuthology.orchestra.run.smithi148.stdout:generic/099      _check_dmesg: something found in dmesg (see /tmp/tmp.GPZijsyn3Pxfstests-dev/results//generic/099.dmesg)
2019-11-06T16:24:42.806 INFO:teuthology.orchestra.run.smithi148.stdout:
2019-11-06T16:24:42.807 INFO:teuthology.orchestra.run.smithi148.stdout:Ran: generic/099
2019-11-06T16:24:42.807 INFO:teuthology.orchestra.run.smithi148.stdout:Failures: generic/099
2019-11-06T16:24:42.807 INFO:teuthology.orchestra.run.smithi148.stdout:Failed 1 of 1 tests
2019-11-06T16:24:42.808 INFO:teuthology.orchestra.run.smithi148.stdout:
2019-11-06T16:24:44.919 DEBUG:teuthology.orchestra.run:got remote process result: 1
2019-11-06T16:24:44.955 INFO:tasks.cephfs_test_runner:test_acls (tasks.cephfs.test_acls.TestACLs) ... ERROR
```

Looks like I need to modify xfstests-dev to test FUSE mounted FS.

jtlayton commented:
The xfstests scripts need to mount and unmount the filesystem. You'll almost certainly need to set up the xfstests config differently for FUSE mounts.

rishabh-d-dave commented:
Testing this PR - http://pulpito.front.sepia.ceph.com/rishabh-2019-11-21_19:20:41-kcephfs-wip-rishabh-pr-29421-distro-basic-smithi/.

rishabh-d-dave commented:
@batrick Tests were successful.

batrick commented on qa/tasks/cephfs/test_acls.py:
```diff
@@ -0,0 +1,26 @@
+            log.info('==================================')
+            log.info('ATM, these tests don\'t work with FUSE mounts.')
+            log.info('==================================')
+            return
```
`self.skipTest("Require kernel client")`

batrick commented on qa/suites/fs/basic_functional/tasks/acls-fuse-client.yaml:
```diff
@@ -0,0 +1,4 @@
+tasks:
+  - cephfs_test_runner:
```
almost forgot, and:

`  fail_on_skip: false`

See also `qa/suites/fs/basic_functional/tasks/client-recovery.yaml`.

batrick commented:
> Testing this PR - http://pulpito.front.sepia.ceph.com/rishabh-2019-11-21_19:20:41-kcephfs-wip-rishabh-pr-29421-distro-basic-smithi/.

This is the kcephfs suite so we would expect it to be the kernel mount in any case :)

rishabh-d-dave commented:
@batrick 
kcephfs tests were successful - http://pulpito.front.sepia.ceph.com/rishabh-2019-12-04_04:37:29-kcephfs-wip-pr-29421-distro-basic-smithi/
